src/enterpriseCompatibilityTest/java/com/gradle/ccud/adapters/enterprise/proxies/BuildScanExtensionProxyTest.java

```diff
@@ -0,0 +1,252 @@
+package com.gradle.ccud.adapters.enterprise.proxies;
+
+import com.gradle.ccud.adapters.reflection.ProxyFactory;
+import com.gradle.scan.plugin.BuildResult;
+import com.gradle.scan.plugin.BuildScanCaptureSettings;
+import com.gradle.scan.plugin.BuildScanDataObfuscation;
+import com.gradle.scan.plugin.BuildScanExtension;
+import com.gradle.scan.plugin.PublishedBuildScan;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class BuildScanExtensionProxyTest extends BaseProxyTest {
+
+    private BuildScanExtension extension;
+    private BuildScanExtensionProxy proxy;
+
+    @BeforeEach
+    void setup() {
+        extension = mock();
+        proxy = ProxyFactory.createProxy(extension, BuildScanExtensionProxy.class);
+    }
+
+    @Test
+    @DisplayName("can set tags using the extension proxy")
+    void testTag() {
+        // when
+        proxy.tag("tag");
+
+        // then
+        verify(extension).tag("tag");
+    }
+
+    @Test
+    @DisplayName("can set custom values using the extension proxy")
+    void testValue() {
+        // when
+        proxy.value("name", "value");
+
+        // then
+        verify(extension).value("name", "value");
+    }
+
+    @Test
+    @DisplayName("can set custom links using the extension proxy")
+    void testLink() {
+        // when
+        proxy.link("name", "value");
+
+        // then
+        verify(extension).link("name", "value");
+    }
+
+    @Test
+    @DisplayName("can set and retrieve the terms of service URL using the extension proxy")
+    void testTermsOfServiceUrl() {
+        // given
+        String url = "https://terms-of-service.com";
+
+        // when
+        proxy.setTermsOfServiceUrl(url);
+
+        // then
+        verify(extension).setTermsOfServiceUrl(url);
+
+        // when
+        when(extension.getTermsOfServiceUrl()).thenReturn(url);
+
+        // then
+        assertEquals(url, proxy.getTermsOfServiceUrl());
+    }
+
+    @Test
+    @DisplayName("can set and retrieve the terms of service agreement using the extension proxy")
```

Comment:
I was wondering why we needed this but then I realized that while this isn't directly used, we customers who extend the functionality of CCUD to customize any of the functionality of the Develocity plugin. For example in `CustomDevelocityConfig` 👍

Reply:
Exactly, that was the idea behind the entire adapter layer.